pkg/registry/apis/query/query.go

```diff
@@ -86,7 +87,8 @@ func (r *queryREST) Connect(ctx context.Context, name string, opts runtime.Objec
 	b := r.builder
 
 	return http.HandlerFunc(func(w http.ResponseWriter, httpreq *http.Request) {
-		ctx, span := b.tracer.Start(httpreq.Context(), "QueryService.Query")
+		ctx := request.WithNamespace(httpreq.Context(), request.NamespaceValue(connectCtx))
+		ctx, span := b.tracer.Start(ctx, "QueryService.Query")
```

Comment:
would it be useful to also add it to the http req context? 

```suggestion
		ctx, span := b.tracer.Start(ctx, "QueryService.Query")
		httpreq = httpreq.WithContext(ctx)
```

Reply:
Yeah, it should probably be the other way round.

Reply:
> it should probably be the other way round

do you mean something like this?
```go
httpreq = httpreq.WithContext(connectCtx)
```

Reply:
No, just that I add first the tracer to the httpreq, and then create the new context. I don't think that it has any implications, as the span is closed after the function anyway, but it looks and feels cleaner.

Reply:
is the namespace only needed by `parseRequest`?

Reply:
For now yes, it's only needed by `parseRequest`.